modules/javafx.base/src/main/java/com/sun/javafx/binding/BidirectionalBinding.java

```diff
@@ -150,7 +151,7 @@ public static BidirectionalBinding bindNumber(DoubleProperty property1, Property
     private static <T extends Number> BidirectionalBinding bindNumber(Property<T> property1, Property<Number> property2) {
         checkParameters(property1, property2);
 
-        final BidirectionalBinding<Number> binding = new TypedNumberBidirectionalBinding<T>(property1, property2);
+        final BidirectionalBinding binding = new TypedNumberBidirectionalBinding<>(property1, property2);
 
         property1.setValue((T)property2.getValue());
         property1.addListener(binding);
```

Comment:
About lines 156-158:

        property1.setValue(property2.getValue());
        property1.addListener(binding);
        property2.addListener(binding);

The second binding added does not validate `property2` anymore as it is an invalidation listener now instead of a change listener.  This means that if `property2` is currently invalid, it will not send an invalidation event to this new listener on the next change.  This small program demonstrates this:

    SimpleDoubleProperty p = new SimpleDoubleProperty(2);

    InvalidationListener invalidationListener = obs -> System.out.println("Invalidated");

    p.addListener(invalidationListener);

    p.setValue(3);
    p.setValue(4);